.github/workflows/twoops-tracker-build-deploy.yml

```
@@ -0,0 +1,70 @@
+name: TwoopsTracker | Build and Deploy
+on:
+  push:
+    branches: [main]
+env:
+  DOKKU_REMOTE_BRANCH: "master"
+  DOKKU_REMOTE_URL: "ssh://dokku@dokku-1.dev.codeforafrica.org/twoopstracker"
+  GIT_PUSH_FLAGS: "--force"
+jobs:
+  build:
+    name: Perform CI Checks
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.10]
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Cache Pants
+      - uses: pantsbuild/actions/init-pants@c0ce05ee4ba288bb2a729a2b77294e9cb6ab66f7
+        # This action bootstraps pants and manages 2-3 GHA caches.
+        # See: github.com/pantsbuild/actions/tree/init-pants/
+        # We use an action SHA because there are no tags yet.
+        with:
+          # cache0 makes it easy to bust the cache if needed
+          # just increase the integer to start with a fresh cache
+          gha-cache-key: cache0-py${{ matrix.python-version }}
+          # If you're not using a fine-grained remote caching service (see https://www.pantsbuild.org/docs/remote-caching),
+          # then you may also want to preserve the local Pants cache (lmdb_store). However this must invalidate for
+          # changes to any file that can affect the build, so may not be practical in larger repos.
+          # A remote cache service integrates with Pants's fine-grained invalidation and avoids these problems.
+          cache-lmdb-store: 'true'  # defaults to 'false'
+          # Note that named_caches and lmdb_store falls back to partial restore keys which
+          # may give a useful partial result that will save time over completely clean state,
+          # but will cause the cache entry to grow without bound over time.
+          # See https://pants.readme.io/docs/using-pants-in-ci for tips on how to periodically clean it up.
+          # Alternatively you change gha-cache-key to ignore old caches.
+      - name: Lint
+        run: |
+          ./pants lint twoops_tracker/py::
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+
+      - name: Build
+        run: |
+          ./pants package twoops_tracker/docker/::
```

```suggestion
      - name: Build
        run: |
          ./pants package twoops_tracker/docker/::

      - name: Login to DockerHub
        uses: docker/login-action@v2
        with:
          username: ${{ secrets.DOCKER_HUB_USERNAME }}
          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
```

Shouldn't we only login to DockerHub after a successful build?